src/jats/importer/jats-dom-parser.ts

```diff
@@ -111,6 +128,227 @@ const marks: MarkRule[] = [
+      const affiliations: string[] = []
+      const corresp: ContributorCorresp[] = []
+
+      const xrefs = element.querySelectorAll('xref')
+      for (const xref of xrefs) {
+        const rid = xref.getAttribute('rid')
+        const type = xref.getAttribute('ref-type')
+        if (!rid) {
+          continue
+        }
+        switch (type) {
+          case 'fn':
+            footnote.push({
+              noteID: rid,
+              noteLabel: xref.textContent?.trim() || '',
+            })
+            break
+          case 'corresp':
+            corresp.push({
+              correspID: rid,
+              correspLabel: xref.textContent?.trim() || '',
+            })
+            break
+          case 'aff':
+            affiliations.push(rid)
+            break
+        }
+      }
+
+      return {
+        id: element.getAttribute('id'),
+        role: 'author',
+        affiliations,
+        corresp,
+        footnote,
+        isCorresponding: element.getAttribute('corresp')
+          ? element.getAttribute('corresp') === 'yes'
+          : undefined,
+        bibliographicName: {
+          given: getTrimmedTextContent(element, 'name > given-names'),
+          family: getTrimmedTextContent(element, 'name > surname'),
+          ObjectType: ObjectTypes.BibliographicName,
+          _id: generateID(ObjectTypes.BibliographicName),
+        },
+        ORCIDIdentifier: getTrimmedTextContent(
+          element,
+          'contrib-id[contrib-id-type="orcid"]'
+        ),
+        priority: parsePriority(element.getAttribute('priority')),
+      }
+    },
+    getContent: () => {
+      return Fragment.from(schema.text('_'))
+    },
+  },
+  {
+    tag: 'affiliations',
+    node: 'affiliations',
+  },
+  {
+    tag: 'aff',
+    node: 'affiliation',
+    context: 'affiliations/',
+    getAttrs: (node) => {
+      const element = node as HTMLElement
+      const getEmail = (element: HTMLElement) => {
+        const email = element.querySelector('email')
+        if (email) {
+          return {
+            href: email.getAttributeNS(XLINK_NAMESPACE, 'href') ?? '',
+            text: email.textContent?.trim() ?? '',
+          }
+        }
+      }
+      const getInstitutionDetails = (element: HTMLElement) => {
+        let department = ''
+        let institution = ''
+        for (const node of element.querySelectorAll('institution')) {
+          const content = node.textContent?.trim()
+          if (!content) {
+            continue
+          }
+          const type = node.getAttribute('content-type')
+          if (type === 'dept') {
+            department = content
+          } else {
+            institution = content
+          }
+        }
+        return { department, institution }
+      }
+      const getAddressLine = (element: HTMLElement, index: number) => {
+        return (
+          getTrimmedTextContent(element, `addr-line:nth-of-type(${index})`) ||
+          ''
+        )
+      }
```

Comment:
Can we move these outside the `getAttrs`  function?

Reply:
Sure